website/blog/2024-03-28-using-post-templates.md

```diff
@@ -0,0 +1,155 @@
+:::
+
+## Adding a template
+
+A recommended approach for adding a post template is to add a service class. For this example, the post type is **Review**, so the WP-CLI command for creating a new service class looks like this:
+
+```bash
+wp boilerplate create service-example --folder=BlockTemplates --file_name=ReviewTemplate
+```
+
+You can call your namespace (folder) **BlockTemplates** or **PostTemplates**, whichever you prefer. If you’re adding a template for another post type, create a separate class, but include it in the same namespace. That way you’ll have a much better overview of the templates in the project.
+
+Here’s an example of how to add a single Quote block as a template:
+
+```php
+/**
+ * Register Review Template.
+ *
+ * @return void
+ */
+public function registerReviewTemplate(): void
+{
+	$reviewPostType = \get_post_type_object(ReviewPostType::POST_TYPE_SLUG);
+
+	$reviewPostType->template = [
+		[
+			'eightshift-boilerplate/quote',
+			[
+				'wrapperSimple' => true,
+				'quoteQuoteParagraphContent' => 'Content placeholder',
+				'quoteQuoteAuthorContent' => 'Author name placeholder',
+			]
+		],
+	];
+}
+```
+
+To assign a post template, you need to get the post type object and then update the `template` property. Alternatively, you can also lock the post template in this method with the `template_lock` property, but it’s much cleaner to do it during custom post type registration.
+
+Don’t forget to register your method with `add_action`:
+
+```php
+\add_action('init', [$this, 'registerReviewTemplate'], 20);
+```
+
+## Building complex layouts
+
+As you can see from the previous example, you can set any attribute you want, including wrapper-related options. The easiest way to add a complex post template is by building it in the Gutenberg editor first, then switching to Code Editor and copying the content. This will have to be rewritten in PHP to make it work, and here’s how to do it. Let’s say the code for the blocks looks like this:
+
+```json
+<!-- wp:eightshift-boilerplate/paragraph {"paragraphParagraphContent":"Placeholder text"} /-->
+
+<!-- wp:eightshift-boilerplate/cta-modal {"ctaModalButtonContent":"Apply now!","ctaModalEndDateHide":true} -->
+<!-- wp:eightshift-boilerplate/heading {"wrapperUseShowControl":false,"wrapperSimpleShowControl":false,"wrapperSimple":true,"wrapperAnchorId":"apply-now","headingHeadingContent":"Application form","headingHeadingSize":"h4:bold","headingAlign":"center"} /-->
+
+<!-- wp:eightshift-forms/forms /-->
+<!-- /wp:eightshift-boilerplate/cta-modal -->
+```
+
+This is how it looks when you rewrite it in PHP.
+
+```php
+$eventPostType->template = [ // array that holds all blocks.
+	[ // array that holds a Paragraph block.
+		'eightshift-boilerplate/paragraph', // full block name.
+		[ // array that holds block attributes.
+			'paragraphParagraphContent' => 'Placeholder text',
+		]
+	],
```

Comment:
Maybe instead of all these comments, extract one block above and just explain how it's structured

Reply:
Not sure about this one, on one hand it does seem a bit cluttered with the comments, but somehow I prefer the idea of having a real example broken down and added comments that explain each part.